mautic/mautic
# Campaign event api fix

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | /
| Related developer documentation PR URL | /
| Issue(s) addressed                     | /



#### Description:

When doing the 4.1.1 release there is a step to test API Library tests on the release. It failed on:
```
1) Mautic\Tests\Api\CampaignsTest::testCampaignContactEditEvent
The response has unexpected status code (500).

Response: {"errors":[{"message":"Call to a member function getTimestamp() on null","code":500,"type":null}],"trace":[{"namespace":"","short_class":"","class":"","type":"","function":"","file":"\/var\/www\/html\/mautic\/app\/bundles\/CampaignBundle\/Model\/SummaryModel.php","line":45,"args":[]},{"namespace":"Mautic\\CampaignBundle\\Model","short_class":"SummaryModel","class":"Mautic\\CampaignBundle\\Model\\SummaryModel","type":"-\u003E","function":"updateSummary","file":"\/var\/www\/html\/mautic\/app\/bundles\/CampaignBundle\/Executioner\/Logger\/EventLogger.php","line":105,"args":[]},{"namespace":"Mautic\\CampaignBundle\\Executioner\\Logger","short_class":"EventLogger","class":"Mautic\\CampaignBundle\\Executioner\\Logger\\EventLogger","type":"-\u003E","function":"persistLog","file":"\/var\/www\/html\/mautic\/app\/bundles\/CampaignBundle\/Executioner\/Scheduler\/EventScheduler.php","line":131,"args":[]},{"namespace":"Mautic\\CampaignBundle\\Executioner\\Scheduler","short_class":"EventScheduler","...

Failed asserting that false is true.

/home/runner/work/api-library/api-library/tests/Api/MauticApiTestCase.php:89
/home/runner/work/api-library/api-library/tests/Api/CampaignsTest.php:436
```

The issue was introduced by https://github.com/mautic/mautic/pull/9471 which was merged in to 4.1.0. 

This PR fixes that error and improves the error messages to be actually helpful. The functional test added covers all error scenarios as well as it tests valid data.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. The problematic API endpoint is covered by the [test](https://github.com/mautic/mautic/pull/10708/files#diff-86af470c80e4d8030634ff501e54d94d133d5006b298add435902cb5081d0557) in this PR so you can poke around that if you want.
3. The fix is in the [Campaign Summary](https://github.com/mautic/mautic/pull/9471) (aggregate stats table) feature so you can test if it works.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10708"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
